File: app/src/main/java/re/snapwi/orientation/io/Joke.java
package re.snapwi.orientation.io;

import android.os.Parcel;
import android.os.Parcelable;

public class Joke implements Parcelable {
  private Value value;

  public String getJoke() {
    return value.getJoke();
  }

  protected Joke(Parcel in) {
    value = (Value) in.readValue(Value.class.getClassLoader());
  }

  @Override public int describeContents() {
    return 0;
  }

  @Override public void writeToParcel(Parcel dest, int flags) {
    dest.writeValue(value);
  }

  @SuppressWarnings("unused")
  public static final Parcelable.Creator<Joke> CREATOR = new Parcelable.Creator<Joke>() {
    @Override public Joke createFromParcel(Parcel in) {
      return new Joke(in);
    }

    @Override public Joke[] newArray(int size) {
      return new Joke[size];
    }
  };
}

File: app/src/main/kotlin/re/snapwi/orientation/io/Joke.kt
package re.snapwi.orientation.io

import android.os.Parcel
import android.os.Parcelable

public class Joke : Parcelable {
  var value: Value? = null

  public constructor(parcel: Parcel) {
    value = parcel.readValue(javaClass<Value>().getClassLoader()) as Value
  }

  public fun getJoke(): String = value?.joke ?: "";

  override fun describeContents(): Int = 0

  override fun writeToParcel(dest: Parcel, flags: Int) {
    dest.writeValue(value)
  }

  companion object {
    public val CREATOR: Parcelable.Creator<Joke> = object : Parcelable.Creator<Joke> {
      override fun createFromParcel(parcel: Parcel): Joke = Joke(parcel)

      override fun newArray(size: Int): Array<Joke?> = arrayOfNulls(size)
    }
  }
}